partiql-spi/src/main/kotlin/org/partiql/spi/function/builtins/DiadicOperator.kt

```diff
@@ -348,116 +371,74 @@ internal abstract class DiadicOperator(
+        // DATE
+        fillTable(PType.INTERVAL_YM, PType.DATE) { lhs, rhs -> getIntervalDateInstance(lhs, rhs) }
+        fillTable(PType.INTERVAL_DT, PType.DATE) { lhs, rhs -> getIntervalDateInstance(lhs, rhs) }
+        fillTable(PType.DATE, PType.INTERVAL_YM) { lhs, rhs -> getDateIntervalInstance(lhs, rhs) }
+        fillTable(PType.DATE, PType.INTERVAL_DT) { lhs, rhs -> getDateIntervalInstance(lhs, rhs) }
+
+        // TIME
+        fillTable(PType.INTERVAL_YM, PType.TIME) { lhs, rhs -> getIntervalTimeInstance(lhs, rhs) }
+        fillTable(PType.INTERVAL_DT, PType.TIME) { lhs, rhs -> getIntervalTimeInstance(lhs, rhs) }
+        fillTable(PType.TIME, PType.INTERVAL_YM) { lhs, rhs -> getTimeIntervalInstance(lhs, rhs) }
+        fillTable(PType.TIME, PType.INTERVAL_DT) { lhs, rhs -> getTimeIntervalInstance(lhs, rhs) }
+
+        // TIMEZ
+        fillTable(PType.INTERVAL_DT, PType.TIMEZ) { lhs, rhs -> getIntervalTimeInstance(lhs, rhs) } // TODO: Create a specific timezone impl
+        fillTable(PType.INTERVAL_YM, PType.TIMEZ) { lhs, rhs -> getIntervalTimeInstance(lhs, rhs) } // TODO: Create a specific timezone impl
+        fillTable(PType.TIMEZ, PType.INTERVAL_YM) { lhs, rhs -> getTimeIntervalInstance(lhs, rhs) } // TODO: Create a specific timezone impl
+        fillTable(PType.TIMEZ, PType.INTERVAL_DT) { lhs, rhs -> getTimeIntervalInstance(lhs, rhs) } // TODO: Create a specific timezone impl
+
+        // TIMESTAMP
+        fillTable(PType.INTERVAL_YM, PType.TIMESTAMP) { lhs, rhs -> getIntervalTimestampInstance(lhs, rhs) }
+        fillTable(PType.INTERVAL_DT, PType.TIMESTAMP) { lhs, rhs -> getIntervalTimestampInstance(lhs, rhs) }
+        fillTable(PType.TIMESTAMP, PType.INTERVAL_YM) { lhs, rhs -> getTimestampIntervalInstance(lhs, rhs) }
+        fillTable(PType.TIMESTAMP, PType.INTERVAL_DT) { lhs, rhs -> getTimestampIntervalInstance(lhs, rhs) }
+
+        // TIMESTAMPZ
+        fillTable(PType.INTERVAL_YM, PType.TIMESTAMPZ) { lhs, rhs -> getIntervalTimestampInstance(lhs, rhs) } // TODO: Create a specific timezone impl
+        fillTable(PType.INTERVAL_DT, PType.TIMESTAMPZ) { lhs, rhs -> getIntervalTimestampInstance(lhs, rhs) } // TODO: Create a specific timezone impl
+        fillTable(PType.TIMESTAMPZ, PType.INTERVAL_YM) { lhs, rhs -> getTimestampIntervalInstance(lhs, rhs) } // TODO: Create a specific timezone impl
+        fillTable(PType.TIMESTAMPZ, PType.INTERVAL_DT) { lhs, rhs -> getTimestampIntervalInstance(lhs, rhs) } // TODO: Create a specific timezone impl
+
+        // INTERVAL
+        fillTable(PType.INTERVAL_YM, PType.INTERVAL_YM) { lhs, rhs -> getIntervalInstance(lhs, rhs) }
+        fillTable(PType.INTERVAL_DT, PType.INTERVAL_YM) { lhs, rhs -> getIntervalInstance(lhs, rhs) }
+        fillTable(PType.INTERVAL_YM, PType.INTERVAL_DT) { lhs, rhs -> getIntervalInstance(lhs, rhs) }
+        fillTable(PType.INTERVAL_DT, PType.INTERVAL_DT) { lhs, rhs -> getIntervalInstance(lhs, rhs) }
```

Comment:
nit: for this section, we can just pass the function as an argument to `fillTable`. For example for 

```Kotlin
fillTable(PType.INTERVAL_YM, PType.DATE) { lhs, rhs -> getIntervalDateInstance(lhs, rhs) }
```


you can just do:

```Kotlin
fillTable(PType.INTERVAL_YM, PType.DATE, ::getIntervalDateInstance)
```